packages/react-components/react-timepicker-compat-preview/stories/TimePicker/TimePickerCustomParsingAndValidation.stories.tsx

```diff
@@ -0,0 +1,85 @@
+import * as React from 'react';
+import { Field, FieldProps, makeStyles } from '@fluentui/react-components';
+import { TimePicker, TimePickerErrorType, TimePickerProps } from '@fluentui/react-timepicker-compat-preview';
+import story from './TimePickerCustomParsingAndValidation.md';
+
+const useStyles = makeStyles({
+  control: {
+    maxWidth: '300px',
+  },
+});
+
+const getErrorMessage = (error?: TimePickerErrorType): FieldProps['validationMessage'] => {
+  switch (error) {
+    case 'invalid-input':
+      return 'Invalid time format. Please use the 24-hour format HH:MM.';
+    case 'out-of-bounds':
+      return 'Time out of the 10:00 to 19:59 range.';
+    case 'required-input':
+      return 'Time is required.';
+    default:
+      return '';
+  }
+};
+
+export const CustomParsingAndValidation = () => {
+  const styles = useStyles();
+
+  const [anchor] = React.useState(new Date('November 25, 2023'));
+
+  const [errorType, setErrorType] = React.useState<TimePickerErrorType>();
+
+  const handleTimeSelect: TimePickerProps['onTimeChange'] = (_ev, data) => {
+    setErrorType(data.errorType);
+
+    if (
+      !data.errorType &&
+      data.selectedTime &&
+      (data.selectedTime.getHours() < 10 || data.selectedTime.getHours() >= 20)
+    ) {
+      setErrorType('out-of-bounds');
+    }
+  };
+
+  const formatTimeStringToDate: TimePickerProps['formatTimeStringToDate'] = (time: string | undefined) => {
```

Comment:
Also just realized that the naming is inaccurate it should rather be `parseTimeStringToDate` - it's reflected in the title of the story too 🤷

Reply:
The naming is because we have a prop pair:
<img width="841" alt="image" src="https://github.com/microsoft/fluentui/assets/28751745/6aeed50b-ba3e-41f6-baa2-d5979377ae61">

Reply:
I renamed formatTimeStringToDate to `parseTimeStringToDate`